Source/CarthageKit/Xcode.swift

```diff
@@ -885,27 +885,36 @@ public func buildScheme(scheme: String, withConfiguration configuration: String,
 			let values = map(sdksByPlatform) { ($0, $1) }
 			return SignalProducer(values: values)
 		}
-		|> flatMap(.Concat) { platform, sdks in
+		|> flatMap(.Concat) { platform, sdks -> SignalProducer<(Platform, [SDK], Bool), CarthageError> in
+			let filterResult = sdkFilter(sdks: sdks, scheme: scheme, configuration: configuration, project: project)
+
+			if let filteredSDKs = filterResult.value where filteredSDKs.isEmpty {
+				return .empty
+			}
+
+			return SignalProducer(result: filterResult.map { filteredSDKs in
+				let isFiltered = filteredSDKs.count != sdks.count
+				return (platform, filteredSDKs, isFiltered)
+			})
+		}
+		|> flatMap(.Concat) { platform, sdks, isFiltered in
 			let folderURL = workingDirectoryURL.URLByAppendingPathComponent(platform.relativePath, isDirectory: true).URLByResolvingSymlinksInPath!
 
-			let sdksToBuild = sdkFilter(sdks: sdks, scheme: scheme, configuration: configuration, project: project)
-			
 			// TODO: Generalize this further?
-			switch sdksToBuild.count {
+			switch sdks.count {
 			case 0:
-				let isMissingSigningIdentities = !sdks.isEmpty
-				let identityAddendum = isMissingSigningIdentities ? " (you're missing one or more signing identities)" : ""
-				fatalError("No valid SDKs found to build\(identityAddendum)")
+				let filterlingAddendum = isFiltered ? " (you may be missing one or more signing identities)" : ""
+				fatalError("No valid SDKs found to build\(filterlingAddendum)")
```

Comment:
This `case` is dead code now, because of [the change above](https://github.com/Carthage/Carthage/pull/798/files#diff-f66644e8cfbcad3c3826395a690ddfbcR891).